internal/pkg/levee/levee.go

```diff
@@ -69,15 +78,53 @@ func run(pass *analysis.Pass) (interface{}, error) {
 	return nil, nil
 }
 
-func reportSourcesReachingSink(conf *config.Config, pass *analysis.Pass, propagations map[*source.Source]propagation.Propagation, sink ssa.Instruction) {
+func reportSourcesReachingSink(conf *config.Config, pass *analysis.Pass, suppressedNodes suppression.ResultType, propagations map[*source.Source]propagation.Propagation, sink ssa.Instruction) {
 	for src, prop := range propagations {
-		if prop.IsTainted(sink) {
+		if prop.IsTainted(sink) && !isSuppressed(sink.Pos(), suppressedNodes, pass) {
 			report(conf, pass, src, sink.(ssa.Node))
 			break
 		}
 	}
 }
 
+func isSuppressed(pos token.Pos, suppressedNodes suppression.ResultType, pass *analysis.Pass) bool {
+	for _, f := range pass.Files {
+		if pos < f.Pos() || f.End() < pos {
+			continue
+		}
+		// astutil.PathEnclosingInterval produces the list of nodes that enclose the provided
+		// position, from the leaf node that directly contains it up to the ast.File node
+		path, _ := astutil.PathEnclosingInterval(f, pos, pos)
+		if len(path) < 2 {
+			return false
+		}
+		// Given the position of a call, path[0] holds the ast.CallExpr and
+		// path[1] holds the ast.ExprStmt. A suppressing comment may be associated
+		// with the name of the function being called (Ident, SelectorExpr), with the
+		// call itself (CallExpr), or with the entire expression (ExprStmt).
+		if ce, ok := path[0].(*ast.CallExpr); ok {
```

Comment:
Can this type assertion fail?  What does it mean if it does?

Reply:
Right now, in my understanding, no. Since sinks are functions and we report on calls to those functions, the first element of `path` should be an `*ast.CallExpr`. I guess I'm just trying to be defensive here and avoid a panic in case those assumptions ever change.

Reply:
Gotcha.  I agree we probably shouldn't panic, but it might be good to add a warning in that unexpected case.  We have a few of those throughout the codebase, e.g. `propagation.go:215: fmt.Printf("unexpected node received: %T %v; please report this issue\n", n, n)`